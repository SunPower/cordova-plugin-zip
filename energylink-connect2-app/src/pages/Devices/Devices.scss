@import '../../../node_modules/@sunpower/theme-dark/styles/variables';

.sunpower-devices {
  .equipment-list {
    max-height: 250px;
    .equipment-piece {
      justify-content: space-between;
    }
    .equipment-piece:after {
      content: '';
      width: 100%;
      height: 1px;
      background-color: $white;
      margin: 5px 0;
    }
    .equipment-piece:last-child:after {
      width: 100%;
      height: 0;
      margin: 0;
    }
  }
}

.inline-loader{
  flex: 0 1 auto;
  flex-direction: column;
  flex-shrink: 0;
  flex-basis: 25%;
  height: 40px;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  display: -ms-flexbox;
  display: flex;
}


.device-prog {
  background-color: $silver-sand4;
  text-align: left;
  padding: $font-size-big $font-size-big $font-size;
  font-family: $font;
  letter-spacing: 0;
  border-radius: 5px;

  &-header {
    color: $white;
    font-size: 18px;
    display: grid;
    grid-template-columns: 70% 1fr;
    margin-bottom: 5px;
  }

  &-title {
    display: grid;
    grid-template-columns: 20% 1fr;
    align-items: center;
  }

  &-status {
    text-align: right;
    padding-right: 10px;
    display: grid;
    align-items: center;
  }
}

.eqs-device{
  border-bottom: 1px solid $silver-sand4;
}
