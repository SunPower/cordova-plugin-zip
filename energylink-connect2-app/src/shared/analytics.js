import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
