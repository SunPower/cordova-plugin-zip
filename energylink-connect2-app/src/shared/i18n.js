import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
