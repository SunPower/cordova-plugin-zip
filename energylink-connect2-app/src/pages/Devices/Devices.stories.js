import React from 'react'
import { Provider } from 'react-redux'
import { storiesOf } from '@storybook/react'

import { configureStore } from 'state/store'

import Devices from '.'

const discoveryInProgress = {
  pvs: {
    serialNumbers: [
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      }
    ]
  },
  devices: {
    candidates: [
      { SERIAL: 'E00110223232323', STATEDESCR: 'NEW' },
      { SERIAL: 'E00110223232324', STATEDESCR: 'OK' }
    ],
    claimingDevices: false,
    claimError: null,
    claimProgress: 0,
    error: null,
    progress: 10,
    discoverComplete: false
  }
}

const discoveryComplete = {
  pvs: {
    serialNumbers: [
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      }
    ]
  },
  devices: {
    candidates: [
      { SERIAL: 'E00110223232323', STATEDESCR: 'OK' },
      { SERIAL: 'E00110223232324', STATEDESCR: 'OK' }
    ],
    claimingDevices: false,
    claimError: null,
    claimProgress: 0,
    error: null,
    progress: 100,
    discoveryComplete: true
  }
}

const discoveryFailed = {
  pvs: {
    serialNumbers: [
      {
        serial_number: 'E00110223232323'
      },
      {
        serial_number: 'E00110223232324'
      }
    ]
  },
  devices: {
    candidates: [
      { SERIAL: 'E00110223232323', STATEDESCR: 'PLC_STATS_ERROR' },
      { SERIAL: 'E00110223232324', STATEDESCR: 'PLC_STATS_ERROR' }
    ],
    claimingDevices: false,
    claimError: null,
    claimProgress: 0,
    error: null,
    progress: 100,
    discoveryComplete: true
  }
}

storiesOf('MI Discovery', module)
  .add('In Progress', () => {
    const { store } = configureStore(discoveryInProgress)

    return (
      <div className="full-min-height pt-20 pb-20">
        <Provider store={store}>
          <Devices />
        </Provider>
      </div>
    )
  })
  .add('Complete - all found', () => {
    const { store } = configureStore(discoveryComplete)

    return (
      <div className="full-min-height pt-20 pb-20">
        <Provider store={store}>
          <Devices />
        </Provider>
      </div>
    )
  })
  .add('Complete - no devices found', () => {
    const { store } = configureStore(discoveryFailed)

    return (
      <div className="full-min-height pt-20 pb-20">
        <Provider store={store}>
          <Devices />
        </Provider>
      </div>
    )
  })
