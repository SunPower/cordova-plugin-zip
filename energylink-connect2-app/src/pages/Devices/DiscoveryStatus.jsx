import React from 'react'
import { is } from 'ramda'
import { useI18n } from 'shared/i18n'
import { either } from 'shared/utils'

function DiscoveryStatus({
  error,
  expected,
  okMICount,
  errMICount,
  claimError,
  claimingDevices,
  claimDevices,
  claimProgress,
  discoveryComplete,
  retryDiscovery,
  cleanAndGoBack,
  areOnboardMetersMissing
}) {
  const t = useI18n()
  if (
    expected === okMICount + errMICount &&
    discoveryComplete &&
    !claimingDevices
  ) {
    if (errMICount > 0) {
      return (
        <>
          <div className="has-text-centered mb-10">
            <span className="has-text-weight-bold mt-20">
              {t('NOT_FOUND_MIS')}
            </span>
          </div>
          <div className="is-flex">
            <button
              className="button is-primary is-uppercase is-outlined is-fullwidth mr-10"
              onClick={cleanAndGoBack}
            >
              {t('GO_BACK')}
            </button>
            <button
              onClick={retryDiscovery}
              className="button is-primary is-uppercase is-fullwidth ml-10"
            >
              {t('RETRY')}
            </button>
          </div>
        </>
      )
    }

    if (error) {
      return (
        <>
          <div className="has-text-centered mb-10">
            <span className="has-text-weight-bold mt-20">
              {t(is(String, error) ? error : 'DEVICE_UNKNOWN_ERROR')}
            </span>
          </div>
          <div className="is-flex">
            <button
              className="button is-primary is-uppercase is-outlined is-fullwidth mr-10"
              onClick={cleanAndGoBack}
            >
              {t('GO_BACK')}
            </button>
            <button
              onClick={retryDiscovery}
              className="button is-primary is-uppercase is-fullwidth ml-10"
            >
              {t('RETRY')}
            </button>
          </div>
        </>
      )
    }

    if (claimError) {
      return (
        <>
          <div>
            <span className="has-text-weight-bold mb-20">
              {t('CLAIM_DEVICES_ERROR', claimError)}
            </span>
          </div>
          <div className="is-flex">
            <button
              className="button is-primary is-outlined is-uppercase is-fullwidth mr-10"
              disabled={claimingDevices}
              onClick={cleanAndGoBack}
            >
              {t('ADD_DEVICES')}
            </button>
            <button
              className="button is-primary is-uppercase is-fullwidth ml-10 claim-devices-button"
              disabled={
                claimingDevices || areOnboardMetersMissing || !discoveryComplete
              }
              onClick={claimDevices}
            >
              {either(claimingDevices, `${claimProgress}%`, t('CLAIM_DEVICES'))}
            </button>
          </div>
        </>
      )
    }

    return (
      <div className="is-flex">
        <button
          className="button is-primary is-outlined is-uppercase is-fullwidth mr-10"
          disabled={claimingDevices}
          onClick={cleanAndGoBack}
        >
          {t('ADD_DEVICES')}
        </button>
        <button
          className="button is-primary is-uppercase is-fullwidth ml-10 claim-devices-button"
          disabled={
            claimingDevices || areOnboardMetersMissing || !discoveryComplete
          }
          onClick={claimDevices}
        >
          {either(claimingDevices, `${claimProgress}%`, t('CLAIM_DEVICES'))}
        </button>
      </div>
    )
  } else {
    return error && discoveryComplete ? (
      <>
        <div className="has-text-centered mb-10">
          <span className="has-text-weight-bold">
            {t(is(String, error) ? error : 'DEVICE_UNKNOWN_ERROR')}
          </span>
        </div>
        <div className="is-flex">
          <button
            className="button is-primary is-uppercase is-outlined is-fullwidth mr-10"
            onClick={cleanAndGoBack}
          >
            {t('GO_BACK')}
          </button>
          <button
            onClick={retryDiscovery}
            className="button is-primary is-uppercase is-fullwidth ml-10"
          >
            {t('RETRY')}
          </button>
        </div>
      </>
    ) : (
      <span className="has-text-weight-bold mb-20">
        {either(
          claimingDevices,
          <div className="inline-buttons">
            <button
              className="button is-primary is-outlined is-uppercase is-paddingless mb-10"
              disabled={claimingDevices}
              onClick={retryDiscovery}
            >
              {t('ADD_DEVICES')}
            </button>
            <button
              className="button is-primary is-uppercase claim-devices-button"
              disabled={
                claimingDevices || areOnboardMetersMissing || !discoveryComplete
              }
              onClick={claimDevices}
            >
              {either(claimingDevices, `${claimProgress}%`, t('CLAIM_DEVICES'))}
            </button>
          </div>,
          t('DISCOVERY_IN_PROGRESS')
        )}
      </span>
    )
  }
}

export default DiscoveryStatus
