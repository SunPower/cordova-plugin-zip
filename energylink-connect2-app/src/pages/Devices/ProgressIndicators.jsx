import React from 'react'
import { prop } from 'ramda'
import { useI18n } from 'shared/i18n'

const ProgressIndicators = ({ progressList }) => {
  const t = useI18n()
  const miIndicators = {
    OK: <span className="is-size-4 mr-10 sp-check has-text-white" />,
    ERROR: <span className="is-size-4 mr-10 sp-hey has-text-primary" />,
    LOADING: (
      <div className="inline-loader">
        <div className="ball-scale-ripple">
          <div />
        </div>
      </div>
    )
  }

  return progressList.map(deviceType => {
    const progr = prop('PROGR', deviceType)
    const type = prop('TYPE', deviceType)
    const nfound = prop('NFOUND', deviceType)

    if (type !== 'MicroInverters') {
      return (
        <div className="device-prog mb-10 mt-10">
          <div className="device-prog-header">
            <div className="device-prog-title">
              <span className="has-text-centered">
                {progr !== '100' ? miIndicators.LOADING : miIndicators.OK}
              </span>
              <span className="pl-10">{t(type)}</span>
            </div>
            <div className="device-prog-status">
              {progr !== '100' ? `${progr}%` : `${nfound} ${t('FOUND')}`}
            </div>
          </div>
        </div>
      )
    }
  })
}

export default ProgressIndicators
