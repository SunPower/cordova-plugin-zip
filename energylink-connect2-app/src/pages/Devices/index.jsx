import Collapsible from 'components/Collapsible'
import { pathOr, propOr, length, filter, propEq } from 'ramda'
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import clsx from 'clsx'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import {
  DISCOVER_COMPLETE,
  FETCH_CANDIDATES_INIT,
  DISCOVER_INIT
} from 'state/actions/devices'
import './Devices.scss'

const microInverterIcon = (
  <span className="sp-inverter mr-20 devices-icon ml-0 mt-0 mb-0" />
)
const meterIcon = <span className="mr-20 sp-meter ml-0 mt-0 mb-0" />
const Icon = (num = 0, max = 0, icon = '') => (
  <div className="is-flex">
    <span className={`${icon} mr-10 ml-0 mt-0 mb-0`} />
    <span className="devices-counter mr-10 ml-0 mt-0 mb-0">{`${num}/${max}`}</span>
  </div>
)
const numberItems = num =>
  num !== 0 && (
    <span className="devices-counter mr-10 ml-0 mt-0 mb-0">{num}</span>
  )

const filterFoundPVS = (SNList, candidatesList) => {
  const okPVS = []
  const nonOkPVS = []
  SNList.forEach(device => {
    try {
      const foundCandidate = candidatesList.find(
        item => item.SERIAL === device.serial_number
      )

      if (foundCandidate && foundCandidate.STATEDESCR.toLowerCase() === 'ok') {
        device.state = foundCandidate.STATEDESCR
        okPVS.push(device)
      } else {
        device.state = foundCandidate.STATEDESCR
        nonOkPVS.push(device)
      }
    } catch (e) {
      console.error('Filtering error', e)
    }
  })

  return { proceed: length(SNList) === length(okPVS), okPVS, nonOkPVS }
}

function mapStateToProps({ inventory, devices, pvs }) {
  const { candidates, found } = devices
  const { serialNumbers } = pvs
  const { bom } = inventory
  const expectedMIs = serialNumbers.map(({ model, ...keepAttrs }) => keepAttrs)
  const { proceed, okPVS, nonOkPVS } = filterFoundPVS(expectedMIs, candidates)

  const inverterCount = filter(propEq('item', 'MODULES'), bom)
  const meterCount = filter(propEq('item', 'METERS'), bom)
  return {
    inventory: {
      inverters: pathOr(0, ['value'], inverterCount),
      meter: pathOr(0, ['value'], meterCount)
    },
    found: {
      ...found,
      proceed,
      inverter: [...okPVS, ...nonOkPVS]
    },
    counts: {
      inverter: {
        okPVSCount: length(okPVS),
        errPVSCount: length(nonOkPVS)
      }
    }
  }
}

const Devices = ({ animationState }) => {
  const { found, counts } = useSelector(mapStateToProps)
  const dispatch = useDispatch()
  const t = useI18n()
  useEffect(() => {
    dispatch(FETCH_CANDIDATES_INIT())
    return () => {
      if (animationState === 'exit') dispatch(DISCOVER_COMPLETE())
    }
  }, [dispatch, animationState])
  return (
    <div className="fill-parent is-flex tile is-vertical has-text-centered sunpower-devices pr-15 pl-15">
      <span className="is-uppercase has-text-weight-bold mb-20" role="button">
        {t('DEVICES')}
      </span>
      <span
        className="is-uppercase mb-20 has-text-primary"
        onClick={() => dispatch(DISCOVER_INIT())}
      >
        {t('RESCAN')}
      </span>
      <div className="pb-15">
        <Collapsible
          title={t('MICRO-INVERTERS')}
          icon={microInverterIcon}
          actions={Icon(
            counts.inverter.okPVSCount,
            length(propOr([], 'inverter', found)),
            'sp-gear'
          )}
        >
          <ul className="equipment-list">
            {propOr([], 'inverter', found).map(elem => {
              return (
                <li
                  className="equipment-piece is-flex flow-wrap tile"
                  key={elem.serial_number}
                >
                  <div className="is-flex is-vertical has-text-white tile">
                    <span>
                      <span className="has-text-weight-bold has-text-white">
                        SN:
                      </span>
                      {elem.serial_number}
                    </span>
                    {elem.model ? (
                      <span>
                        {elem.model} <span className="sp-pencil is-gray" />
                      </span>
                    ) : (
                      ''
                    )}
                  </div>
                  <span
                    className={clsx('is-size-4 mr-10', {
                      'sp-check has-text-white': elem.state === 'OK',
                      'sp-hey has-text-primary': elem.state !== 'OK'
                    })}
                  />
                </li>
              )
            })}
          </ul>
        </Collapsible>
      </div>
      <div className="pb-15">
        <Collapsible
          title={t('METERS')}
          actions={numberItems(length(propOr([], 'power meter', found)))}
          icon={meterIcon}
        >
          <ul className="equipment-list">
            {propOr([], 'power meter', found).map(elem => {
              return (
                <li className="equipment-piece is-flex flow-wrap tile">
                  <div className="is-flex is-vertical has-text-white tile">
                    <span>
                      <span className="has-text-weight-bold has-text-white">
                        SN:
                      </span>
                      {elem.SERIAL}
                    </span>
                  </div>
                </li>
              )
            })}
          </ul>
        </Collapsible>
      </div>
      <Link
        className="button is-outlined is-primary is-uppercase is-paddingless ml-75 mr-75 mb-10"
        to={paths.PROTECTED.SN_LIST.path}
      >
        {t('ADD-DEVICES')}
      </Link>
      {found.proceed ? (
        <Link
          className="button is-primary is-uppercase is-paddingless ml-75 mr-75"
          to={paths.PROTECTED.INSTALL_SUCCESS.path}
        >
          {t('DONE')}
        </Link>
      ) : (
        <span className="has-text-weight-bold mb-20">
          {t('DEVICES_NOT_FOUND')}
        </span>
      )}
    </div>
  )
}

export default Devices
