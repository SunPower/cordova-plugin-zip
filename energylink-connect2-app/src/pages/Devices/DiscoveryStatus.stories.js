import { action } from '@storybook/addon-actions'
import { storiesOf } from '@storybook/react'
import { clone } from 'ramda'
import React from 'react'

import DiscoveryStatus from './DiscoveryStatus'

const okMI = [
  {
    serial_number: 'E00110223232323',
    indicator: 'MI_OK',
    STATEDESCR: 'OK',
    SERIAL: 'E00110223232323'
  },
  {
    serial_number: 'E00110223232324',
    indicator: 'MI_OK',
    STATEDESCR: 'OK',
    SERIAL: 'E00110223232324'
  }
]

const props = {
  discoveryComplete: false,
  error: '',
  expected: 2,
  okMICount: okMI.length,
  errMICount: 0,
  claimError: 0,
  claimingDevices: false,
  claimDevices: action('claim devices'),
  claimProgress: 0,
  retryDiscovery: action('retry discovery')
}

storiesOf('Devices - DiscoveryStatus', module)
  .add('When Discovering', () => {
    return (
      <div className="full-min-height">
        <DiscoveryStatus {...props} />
      </div>
    )
  })
  .add('When discovery complete', () => {
    const p2 = clone(props)
    p2.discoveryComplete = true
    return (
      <div className="full-min-height">
        <DiscoveryStatus {...p2} />
      </div>
    )
  })
  .add('When discovery complete and has errors', () => {
    const p2 = clone(props)
    p2.discoveryComplete = true
    p2.error = 'Something happened with the request'
    return (
      <div className="full-min-height">
        <DiscoveryStatus {...p2} />
      </div>
    )
  })
  .add('When claiming devices', () => {
    const p2 = clone(props)
    p2.claimingDevices = true
    p2.claimProgress = 27
    return (
      <div className="full-min-height">
        <DiscoveryStatus {...p2} />
      </div>
    )
  })
  .add('When claiming has errors', () => {
    const p2 = clone(props)
    p2.claimingDevices = false
    p2.discoveryComplete = true
    p2.claimProgress = 27
    p2.claimError = 'UNK'
    return (
      <div className="full-min-height">
        <DiscoveryStatus {...p2} />
      </div>
    )
  })
