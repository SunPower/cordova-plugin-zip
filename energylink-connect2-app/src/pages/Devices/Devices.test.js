import React from 'react'
import * as ReactDOM from 'react-dom'
import * as i18n from 'shared/i18n'
import Devices from '.'

let component
const state = {
  pvs: {
    serialNumbers: [
      {
        serial_number: 'E00110223232323',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      },
      {
        serial_number: 'E00110223232324',
        indicator: 'MI_OK',
        STATEDESCR: 'OK'
      }
    ]
  },
  devices: {
    candidates: [
      { SERIAL: 'E00110223232323', STATEDESCR: 'OK' },
      { SERIAL: 'E00110223232324', STATEDESCR: 'OK' }
    ],
    claimingDevices: false,
    claimError: null,
    claimProgress: 0,
    error: null,
    progress: 10,
    discoverComplete: false
  }
}

describe('Devices page', () => {
  beforeEach(() => {
    jest.spyOn(i18n, 'useI18n').mockImplementation(() => key => key)
    jest
      .spyOn(ReactDOM, 'createPortal')
      .mockImplementation(() => jest.fn((element, node) => element))
    component = mountWithProvider(<Devices />)(state)
  })

  test('Renders correctly', () => {
    expect(component).toMatchSnapshot()
  })
})
