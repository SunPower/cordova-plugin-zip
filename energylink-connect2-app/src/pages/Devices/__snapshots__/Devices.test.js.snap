// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices page Renders correctly 1`] = `
Object {
  "component": <BrowserRouter
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Provider
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <Devices>
          <div
            className="fill-parent is-flex tile is-vertical has-text-centered sunpower-devices pr-15 pl-15"
          >
            <span
              className="is-uppercase has-text-weight-bold mb-20"
              role="button"
            >
              DEVICES
            </span>
            <div
              className="pb-15"
            >
              <Collapsible
                actions={
                  <div>
                    <span
                      className="devices-counter mr-10 ml-0 mt-0 mb-0"
                    >
                      2/2
                    </span>
                  </div>
                }
                expanded={true}
                icon={
                  <span
                    className="sp-inverter mr-20 devices-icon ml-0 mt-0 mb-0"
                  />
                }
                title="MICROINVERTERS"
              >
                <div
                  className="collapsible"
                >
                  <div
                    className="collapsible-header"
                    onClick={[Function]}
                  >
                    <div
                      className="collapsible-title"
                    >
                      <span>
                        <span
                          className="sp-inverter mr-20 devices-icon ml-0 mt-0 mb-0"
                        />
                      </span>
                      <span
                        className="has-text-weight-bold"
                      >
                        MICROINVERTERS
                      </span>
                    </div>
                    <div
                      className="collapsible-actions"
                    >
                      <div>
                        <span
                          className="devices-counter mr-10 ml-0 mt-0 mb-0"
                        >
                          2/2
                        </span>
                      </div>
                    </div>
                    <div
                      className="collapsible-trigger"
                    >
                      <div
                        className="chevron down"
                      >
                        <span
                          className="sp-chevron-up"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="collapsible-content expanded"
                  >
                    <ul
                      className="equipment-list"
                    >
                      <li
                        className="equipment-piece is-flex flow-wrap tile"
                        key="E00110223232323"
                      >
                        <div
                          className="is-flex is-vertical tile"
                        >
                          <span>
                            <span
                              className="has-text-weight-bold has-text-white"
                            >
                              SN:
                            </span>
                            <span
                              className="has-text-white"
                            >
                              E00110223232323
                            </span>
                            <span
                              className="has-text-weight-bold has-text-white ml-10"
                            />
                          </span>
                          <span>
                            MISTATE_MI_OK
                          </span>
                        </div>
                        <div
                          className="is-flex mi-indicator"
                          onClick={[Function]}
                        >
                          <span
                            className="is-size-4 mr-10 sp-check has-text-white"
                          />
                        </div>
                      </li>
                      <li
                        className="equipment-piece is-flex flow-wrap tile"
                        key="E00110223232324"
                      >
                        <div
                          className="is-flex is-vertical tile"
                        >
                          <span>
                            <span
                              className="has-text-weight-bold has-text-white"
                            >
                              SN:
                            </span>
                            <span
                              className="has-text-white"
                            >
                              E00110223232324
                            </span>
                            <span
                              className="has-text-weight-bold has-text-white ml-10"
                            />
                          </span>
                          <span>
                            MISTATE_MI_OK
                          </span>
                        </div>
                        <div
                          className="is-flex mi-indicator"
                          onClick={[Function]}
                        >
                          <span
                            className="is-size-4 mr-10 sp-check has-text-white"
                          />
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </Collapsible>
              <ProgressIndicators
                progressList={Array []}
              />
            </div>
            <DiscoveryStatus
              claimDevices={[Function]}
              claimError={null}
              claimProgress={0}
              claimingDevices={false}
              cleanAndGoBack={[Function]}
              errMICount={0}
              error={null}
              expected={2}
              okMICount={2}
              retryDiscovery={[Function]}
            >
              <span
                className="has-text-weight-bold mb-20"
              >
                DISCOVERY_IN_PROGRESS
              </span>
            </DiscoveryStatus>
          </div>
        </Devices>
      </Provider>
    </Router>
  </BrowserRouter>,
  "store": Object {
    "clearActions": [Function],
    "dispatch": [Function],
    "getActions": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
}
`;
